src/FsAutoComplete/CodeFixes/AdjustConstant.fs

```diff
@@ -0,0 +1,1670 @@
+module FsAutoComplete.CodeFix.AdjustConstant
+
+open System
+open FsToolkit.ErrorHandling
+open FsAutoComplete.CodeFix.Types
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+open FSharp.Compiler.Syntax
+open System.Runtime.CompilerServices
+open FSharp.Compiler.Text.Range
+open Microsoft.FSharp.Core.LanguagePrimitives
+
+/// If `true`: enable `debugFix`es which show parsed `XXXConstant`s.
+///
+/// Note: As constant, because F# doesn't have `#define`
+[<Literal>]
+let private DEBUG = false
+
+let inline private unreachable () = invalidOp "unreachable"
+
+/// Returns `SynConst` and its range at passed `pos`
+///
+/// Note:
+/// When `SynConst.Measure`:
+/// * returns contained constant when `pos` inside contained constant
+/// * otherwise `SynConst.Measure` when on other parts of `Measure` constant (`<km>`)
+///
+/// Note:
+/// Might be erroneous Constant -> containing `value` is then default (`0`).
+/// Check by comparing returned range with existing Diagnostics.
+let private tryFindConstant ast pos =
+  let rec findConst range constant =
+    match constant with
+    | SynConst.Measure(constant, constantRange, _) when rangeContainsPos constantRange pos ->
+      findConst constantRange constant
+    | _ -> (range, constant)
+
+  SyntaxTraversal.Traverse(
+    pos,
+    ast,
+    { new SyntaxVisitorBase<_>() with
+        member _.VisitExpr(_, _, defaultTraverse, expr) =
+          match expr with
+          //                                     without: matches when `pos` in comment after constant
+          | SynExpr.Const(constant, range) when rangeContainsPos range pos -> findConst range constant |> Some
+          | _ -> defaultTraverse expr
+
+        member _.VisitEnumDefn(_, cases, _) =
+          cases
+          |> List.tryPick (fun (SynEnumCase(valueExpr = expr)) ->
+            let rec tryFindConst (expr: SynExpr) =
+              match expr with
+              | SynExpr.Const(constant, range) when rangeContainsPos range pos -> findConst range constant |> Some
+              | SynExpr.Paren(expr = expr) -> tryFindConst expr
+              | SynExpr.App(funcExpr = funcExpr) when rangeContainsPos funcExpr.Range pos -> tryFindConst funcExpr
+              | SynExpr.App(argExpr = argExpr) when rangeContainsPos argExpr.Range pos -> tryFindConst argExpr
+              | _ -> None
+
+            tryFindConst expr)
+
+        member _.VisitPat(_, defaultTraverse, synPat) =
+          match synPat with
+          | SynPat.Const(constant, range) when rangeContainsPos range pos -> findConst range constant |> Some
+          | _ -> defaultTraverse synPat }
+  )
+
+/// Computes the absolute of `n`
+///
+/// Unlike `abs` or `Math.Abs` this here handles `MinValue` and does not throw `OverflowException`.
```

Comment:
This is a wonderful comment - it answered the questions I had as soon as I read the module name :)